server/src/main/java/io/spine/server/command/CommandHandlingEntity.java

```diff
@@ -172,4 +176,70 @@ protected ValueMismatch expectedNotEmpty(String expected) {
     protected ValueMismatch unexpectedValue(String expected, String actual, String newValue) {
         return StringMismatch.unexpectedValue(expected, actual, newValue, versionNumber());
     }
+
+    /**
+     * A object which performs filtering of the events emitted by a
+     * {@link CommandHandlingEntity command handling entity}.
+     */
+    protected static class EventFilter {
```

Comment:
As long as it contains another class (i.e. an `enum` declaration) I'd move it to the upper level for it to become a standalone class.

Then I'd review the method visibility. Right now (while it's still a `protected` nested) its `public static` methods look confusing. Let's see how they will look after the class is extracted.